File: Greedy/Cpp/10610.cpp
// 30
#include <algorithm>
#include <iostream>
#include <string>
using namespace std;

int main() {
    int sum = 0;
    bool mul10 = false;
    string input;
    cin >> input;
    for (int i = 0; i < input.length(); i++) {
        if (input[i] == '0') mul10 = true;
        sum += input[i] - '0';
    }
    if (mul10 == false || sum % 3 != 0)
        cout << -1;
    else {
        sort(input.begin(), input.end(), greater<char>());
        cout << input;
    }
}

File: Greedy/Python/10610.py
string = input()
sum = 0
mul10 = False
for s in string:
    a = int(s)
    if a == 0:
        mul10 = True
    sum += a
if sum % 3 != 0 or mul10 == False:
    print(-1)
else:
    string = sorted(string, reverse=True)
    string = ''.join(string)
    print(string)
